### PR TITLE
[Chaos] Track regression in terms of recovery timing

### PR DESCRIPTION
**Pull request by chaitanyaenr**
This commit:
    - Enables flagging builds on recovery time regression. Regression times are
      captured from multiple runs of the scenario and failed if expected recovery
      time exceeds +5% threshold when compared to the peak time observed in the sample
      data. Components included in this PR: Etcd, Prometheus and kube-apiserver.
    - Adds disruption scenario for Kube-apiserver.

### CODE REVIEW

File: ci-operator/step-registry/redhat-chaos/pod-scenarios/kube-apiserver-disruption/OWNERS [hunk 1]
```diff
@@ -0,0 +1,10 @@
+approvers:
```
Comment (paigerube14) [inline ci-operator/step-registry/redhat-chaos/pod-scenarios/kube-apiserver-disruption/OWNERS:1]: lets use a symbolic link to the owners file in the base redhat-chaos folder like: https://github.com/openshift/release/blob/master/ci-operator/step-registry/redhat-chaos/pod-scenarios/etcd-disruption/OWNERS

Comment (chaitanyaenr) [reply ci-operator/step-registry/redhat-chaos/pod-scenarios/kube-apiserver-disruption/OWNERS]: Done, thanks.

**Comment by chaitanyaenr** [comment]
/pj-rehearse periodic-ci-redhat-chaos-prow-scripts-main-4.17-nightly-krkn-hub-tests-417 periodic-ci-redhat-chaos-prow-scripts-main-4.16-nightly-krkn-hub-tests-416

**Comment by chaitanyaenr** [comment]
/pj-rehearse periodic-ci-redhat-chaos-prow-scripts-main-4.17-nightly-krkn-hub-tests-417 periodic-ci-redhat-chaos-prow-scripts-main-4.16-nightly-krkn-hub-tests-416

**Comment by chaitanyaenr** [comment]
/pj-rehearse periodic-ci-redhat-chaos-prow-scripts-main-4.16-nightly-krkn-hub-tests-416

**Comment by chaitanyaenr** [comment]
/pj-rehearse periodic-ci-redhat-chaos-prow-scripts-main-4.17-nightly-krkn-hub-tests-417 periodic-ci-redhat-chaos-prow-scripts-main-4.16-nightly-krkn-hub-tests-416

**Comment by chaitanyaenr** [comment]
@paigerube14 @shahsahil264 @tsebastiani 4.17 run is GREEN! PTAL when you get time.

Hitting the same issue multiple times with 4.16 random-system-pods scenario:
```
 2024-07-18 23:42:13,657 [ERROR] scenario exception: unrecovered pods: Thread pool did not shutdown correctly,aborting.
Exception: 
2024-07-18 23:42:13,660 [ERROR] scenario: /tmp/pod_scenario.yaml failed with exception: <class 'Exception'> file: /home/krkn/kraken/kraken/plugins/__init__.py line: 277 
```

**Comment by chaitanyaenr** [comment]
/pj-rehearse periodic-ci-redhat-chaos-prow-scripts-main-4.16-nightly-krkn-hub-tests-416

**Comment by chaitanyaenr** [comment]
/pj-rehearse periodic-ci-redhat-chaos-prow-scripts-main-4.16-nightly-krkn-hub-tests-416

**Comment by chaitanyaenr** [comment]
/pj-rehearse periodic-ci-redhat-chaos-prow-scripts-main-4.16-nightly-krkn-hub-tests-416

**Comment by chaitanyaenr** [comment]
/pj-rehearse periodic-ci-redhat-chaos-prow-scripts-main-4.16-nightly-krkn-hub-tests-416

**Comment by chaitanyaenr** [comment]
/pj-rehearse periodic-ci-redhat-chaos-prow-scripts-main-4.16-nightly-krkn-hub-tests-416

**Comment by chaitanyaenr** [comment]
/pj-rehearse periodic-ci-redhat-chaos-prow-scripts-main-4.16-nightly-krkn-hub-tests-416

**Comment by chaitanyaenr** [comment]
/pj-rehearse periodic-ci-redhat-chaos-prow-scripts-main-4.16-nightly-krkn-hub-tests-416

**Review by shahsahil264 — APPROVED** [review]
LGTM

**Pull request merged**